src/utils/clients.ts

```diff
@@ -25,15 +25,22 @@
 // Reset does not happen by actually resending all the events,
 // the API has a special endpoint that first sets all clients as offline.
 
-import { IncomingMessage } from 'http';
+import * as _ from 'lodash';
 import { setTimeout } from 'timers/promises';
 import { Logger } from 'winston';
 
 import { apiKey, captureException } from './index';
-import { request } from './request';
+import { request, Response } from './request';
+import { intVar } from './config';
 
 const BALENA_API_HOST = process.env.BALENA_API_HOST!;
 const REQUEST_TIMEOUT = 60000;
+// As of writing this, using a chunk of 8000 62-char UUIDs results a content-length
+// that is bellow the 512KiB threshold that would trigger a 413 http error.
+const API_DEVICE_STATE_POST_BATCH_SIZE = intVar(
+	'API_DEVICE_STATE_POST_BATCH_SIZE',
+	8000,
```

Comment:
Did you test this limit btw @thgreasi ?

Reply:
Yes, I tested it on my BoB @Page-, increasing this till I got a 413. Mentioned the interesting part of my findings in the PR description. 8060 also worked, but I thought I should better be not use the edge value (otherwise we could have a formula to calculate this number).

Reply:
I'd actually missed that, and yeah I definitely prefer staying well within a safe margin so lgtm